src/ITensorsExtensions/ITensorsExtensions.jl

```diff
@@ -57,7 +59,13 @@ function map_itensor(
 )
   USV = svd(A, lind; kwargs...)
   U, S, V, spec, u, v = USV
-  S = map_diag(s -> f(s + regularization), S)
+
+  if !isnothing(regularization)
+    S = map_diag(s -> f(s + regularization), S)
+  else
+    S = map_diag(s -> f(s), S)
+  end
```

Comment:
I think a nicer design would be:
```julia
f = isnothing(regularization) ? f : f ∘ Base.Fix2(+, regularization)
S = map_diag(f, S)
```
though honestly I think it would be better to remove `regularization` from this function and just expect users to change the mapping function themselves as needed.

Reply:
Yeah that seems sensible to say that the `regularization` should be encoded in `f` already.